src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -1,272 +1,407 @@
+                        }
+                        else if (d.Value == ConstantValue.Null)
+                        {
+                            addConjunct(ref this, _localRewriter.MakeNullCheck(d.Syntax, input, input.Type.IsNullableType() ? BinaryOperatorKind.NullableNullEqual : BinaryOperatorKind.Equal));
+                        }
+                        else
+                        {
+                            var systemObject = _factory.SpecialType(SpecialType.System_Object);
+                            addConjunct(ref this, _localRewriter.MakeEqual(_factory.Literal(d.Value, input.Type), input));
+                        }
+                        break;
+                    case BoundDagFieldEvaluation f:
+                        {
+                            var field = f.Field;
+                            field = field.TupleUnderlyingField ?? field;
+                            var outputTemp = new BoundDagTemp(f.Syntax, field.Type, f, 0);
+                            var output = _tempAllocator.GetTemp(outputTemp);
+                            _sideEffectBuilder.Add(_factory.AssignmentExpression(output, _factory.Field(input, field)));
+                        }
+                        break;
+                    case BoundDagPropertyEvaluation p:
+                        {
+                            var property = p.Property;
+                            property = property.TupleUnderlyingProperty ?? property;
+                            var outputTemp = new BoundDagTemp(p.Syntax, property.Type, p, 0);
+                            var output = _tempAllocator.GetTemp(outputTemp);
+                            _sideEffectBuilder.Add(_factory.AssignmentExpression(output, _factory.Property(input, property)));
+                        }
+                        break;
+                    case BoundDagDeconstructEvaluation d:
                         {
-                            var value = ((BoundLiteral)test).ConstantValue;
-                            return value.IsBoolean && value.BooleanValue;
+                            var method = d.DeconstructMethod;
+                            var refKindBuilder = ArrayBuilder<RefKind>.GetInstance();
+                            var argBuilder = ArrayBuilder<BoundExpression>.GetInstance();
+                            BoundExpression receiver;
+                            void addArg(RefKind refKind, BoundExpression expression)
+                            {
+                                refKindBuilder.Add(refKind);
+                                argBuilder.Add(expression);
+                            }
+                            Debug.Assert(method.Name == "Deconstruct");
+                            int extensionExtra;
+                            if (method.IsStatic)
+                            {
+                                Debug.Assert(method.IsExtensionMethod);
+                                receiver = _factory.Type(method.ContainingType);
```

Comment:
Isn't this the static class where the method lives? Or is this the receiver in expanded form?

Reply:
The compiler always works with extension methods in expanded form. Reduced form is only used in the public API.